javalib/src/main/scala/java/time/YearMonth.scala

```diff
@@ -0,0 +1,246 @@
+package java.time
+
+import java.time.temporal._
+
+/** Created by alonsodomin on 25/12/2015. */
+final class YearMonth private (year: Int, month: Int)
+    extends TemporalAccessor with Temporal with TemporalAdjuster
+    with Comparable[YearMonth] with java.io.Serializable {
+
+  import Preconditions._
+  import ChronoField._
+  import ChronoUnit._
+
+  requireDateTime(year >= Year.MIN_VALUE && year <= Year.MAX_VALUE,
+      s"Invalid year: $year")
+  requireDateTime(month >= 1 && month <= 12, s"Invalid month: $month")
+
+  def isSupported(field: TemporalField): Boolean = field match {
+    case _: ChronoField =>
+      field == YEAR || field == YEAR_OF_ERA || field == MONTH_OF_YEAR ||
+        field == PROLEPTIC_MONTH || field == ERA
+
+    case null => false
+    case _    => field.isSupportedBy(this)
+  }
+
+  def isSupported(unit: TemporalUnit): Boolean = unit match {
+    case _: ChronoUnit =>
+      unit == MONTHS || unit == YEARS || unit == DECADES || unit == CENTURIES ||
+        unit == MILLENNIA || unit == ERAS
+
+    case null => false
+    case _    => unit.isSupportedBy(this)
+  }
+
+  override def range(field: TemporalField): ValueRange = field match {
+    case YEAR_OF_ERA =>
+      if (year < 1) ValueRange.of(1, Year.MAX_VALUE + 1)
+      else ValueRange.of(1, Year.MAX_VALUE)
+
+    case _ => super.range(field)
+  }
```

Comment:
You have already overriden this in `Year` and you will probably end up implementing it in every class that supports it. You should try to put it directly in the implementation of `TemporalAccessor.range`.